File: src/0001-0100/020 - Valid Parentheses/valid_parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for i in s:
            if i in ["(", "[", "{"]:
                stack.append(i)
            elif i == ")" and (stack == [] or stack.pop() != "("):
                return False
            elif i == "]" and (stack == [] or stack.pop() != "["):
                return False
            elif i == "}" and (stack == [] or stack.pop() != "{"):
                return False
        return stack == []

File: src/0001-0100/020 - Valid Parentheses/ValidParentheses.java
import java.util.Stack;

class Solution {
  public boolean isValid(String s) {
    if (s.length() < 2)
      return false;

    Stack<Character> stack = new Stack<>();

    for (char ch : s.toCharArray()) {
      if (ch == '(' || ch == '[' || ch == '{')
        stack.push(ch);
      else if (ch == ')' && (stack.isEmpty() || stack.pop() != '('))
        return false;
      else if (ch == ']' && (stack.isEmpty() || stack.pop() != '['))
        return false;
      else if (ch == '}' && (stack.isEmpty() || stack.pop() != '{'))
        return false;
    }

    return stack.isEmpty();
  }
}